src/color.rs

```diff
@@ -1289,3 +1287,61 @@ where
         alpha,
     ))
 }
+
+/// Try to parse the components and alpha with the legacy syntax, but also allow
+/// the [color-4] syntax if that fails.
+/// https://drafts.csswg.org/css-color-4/#color-syntax-legacy
+pub fn parse_legacy_components<'i, 't, P, F1, F2, F3, R1, R2, R3>(
```

Comment:
Maybe `parse_components_allow_legacy` would be a better name for this?

(But then again we could have a single `parse_components` with `allow_legacy: bool`). I assume this makes it easier to handle `none` in the future somehow tho, so this is fine as an interim step.

Reply:
Yes, adding `none` complicates things, because the legacy syntax doesn't allow `none`.  But also, because if the function supports legacy, then the newer syntax for that same function doesn't allow `none` either.  So adding the `allow_legacy` turns into an if explosion.